src/Native/configure.cmake

```diff
@@ -47,6 +48,30 @@ check_cxx_source_compiles(
     "
     HAVE_GNU_STRERROR_R)
 
+
+if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+    set (CMAKE_REQUIRED_LIBRARIES rt)
+endif ()
+
+check_cxx_source_runs(
+    "
+    #include <sys/mman.h>
+    #include <fcntl.h>
+    #include <unistd.h>
+
+    int main()
+    {
+        int fd = shm_open(\"/corefx_configure_shm_open\", O_CREAT | O_RDWR, 0777);
+        shm_unlink(\"/corefx_configure_shm_open\");
+
+        // NOTE: PROT_EXEC and MAP_PRIVATE don't work with shm_open on at least current version OS X.
+        return mmap(nullptr, 1, PROT_EXEC, MAP_PRIVATE, fd, 0) == MAP_FAILED ? -1 : 0;
```

Comment:
IIRC, there was also a significant issue with regards to being able to successfully do ftruncate on the fd from shm_open on OS X.